syntax = "proto3";

import "model.proto";

option java_multiple_files = true;
option java_package = "de.ilem0n.udaconnect.rpc";
option java_outer_classname = "ConnectionsService";

package connections;

// connections service
service Connections {
  // provides connections for a give person by person id
  rpc GetConnections (ConnectionsRequest) returns (ConnectionsReply) {}
}

// The request message containing the person id
message ConnectionsRequest {
  int32 personId = 1;
}


// The response message containing the connections for the desired person
message ConnectionsReply {
  int32 personId = 1;
  repeated ConnectionDTO connections = 2;
}